vita3k/modules/SceAppUtil/SceAppUtil.cpp

```diff
@@ -363,7 +363,7 @@ static SceInt32 SafeMemory(EmuEnvState &emuenv, const void *buf, SceSize bufSize
         write_file(fd, safe_mem.data(), SCE_APPUTIL_SAFEMEMORY_MEMORY_SIZE, emuenv.io, export_name);
         close_file(emuenv.io, fd, export_name);
     } else
-        memcpy(&buf, &safe_mem[offset], bufSize);
+        memcpy(const_cast<void *>(buf), &safe_mem[offset], bufSize);
```

Comment:
Can you remove the `const_cast` and instead remove the `const ` in the function declaration?

Reply:
no is const, keep it

Reply:
If you need to const cast, then it's not const, as you can see this function is writing to the location this pointer points to.

Reply:
save is const, load, i hgave code like that for keep const in save

Reply:
Yes, save is const but load is not and this function does both so do not make it const. `const_cast` should only be used really exceptionally.

Reply:
it should be changed on function declaration as macdu said. Gotta remote const_cast.

Reply:
> it should be changed on function declaration as macdu said. Gotta remote const_cast.

what function declared ? don't touch on header of official function please 💀

Reply:
We are not talking about the header of the official functions which are correct, we are talking about the static function `SafeMemory`...

Reply:
noworks for me, remove it get crash on load save in escape plain

Reply:
Escape plain using Save mode for load save, and after in everyu chapter finished, no works without const

Reply:
sceAppUtilSaveSafeMemory has a const void* so i have to use const_cast there or not make the buffer not const on that function, which one is prefered?

Reply:
Const cast is good for me, is not possible remove const for save

Reply:
Yes, use the `const_cast` in `sceAppUtilSaveSafeMemory`, it is better.

Reply:
should be good now